In [14]:
import numpy as np
def rule1(dA,c):
    dQ1=(dA*c)
    return dQ1

In [15]:
def rule2(dA,A,m,c):
    dQ2=c*m*dA*A**(m-1)
    return dQ2

In [16]:
def rule3(dA,dB):
    dQ3=(np.sqrt(dA**2+dB**2))
    return dQ3

In [17]:
def rule4(m,n,dA,A,dB,B,Q):
    dQ4=Q*np.sqrt(((m*dA)/A)**2+((n*dB)/B)**2)
    return dQ4

In [28]:
#Calculation for Method 2
def rule4(Q,m,dA,A,n,dB,B):
    dQ4=Q*np.sqrt((((m*dA)/A)**2)+(((n*dB)/B)**2))
    return dQ4
Q=3.54
m=1
n=1
dA=0.004
A=1.591
dB=0.0001
B=0.989
dQ4=rule4(Q,m,dA,A,n,dB,B)
print("dQ = ",dQ4)

dQ =  0.008907257597900108


The uncertainty for method 2 in the lab was 0.009 m/s and the uncertainty for method 2 here was 0.009 m/s. These two computations are equivalent. The only difference is one was calculated by hand and the other was calculated by programming. 

In [29]:
x=np.array([1.1,1.3,1.4,0.9,0.95,1.05])
xaverage=np.average(x)
xstd=np.std(x)
xerror=np.std(x)/np.sqrt(6)
print("Average =",xaverage)
print("Standard Deviation =",xstd)
print("Error in x =", xerror)

Average = 1.1166666666666667
Standard Deviation = 0.17950549357115014
Error in x = 0.073282810879294


The average of the data set was found to be 1.12.
The standard deviation of the data set was found to be 0.180.
The error in the mean was found to be 0.0733.

Rule 3:
$\delta Q= \sqrt{(\delta A)^2 +(\delta B)^2}$


$\delta vi= vi*\sqrt{(\frac{\delta x}{x})^2 +(\frac{\delta h}{h})^2}$